### ISSUE TITLE
link-name rule fails when it contains landmarks

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#1461]
Found the following example which seemed to fail axe-core:

```html
<a href="...">
  <article><div>
    <div><img src="..." alt=""></div>
    <p class="card-description">Some card</p>
  </div></article>
</a>
```

As far as I can tell this link should and does not have an accessible name, but in a quick test with VO that doesn't seem to be a problem.

axe-core: 3.2.2

**Comment by drewlee** [comment]
We've noticed this issue as well after upgrading to v3.2.2. Links which had previously passed are now getting flagged as violations. Are there any workarounds aside from excluding the element or rule from the audit?

**Comment by WilcoFiers** [comment]
We've done a little more testing. Here are the results so far:

| AT + UA     | method    | announcement |
|-------------|-----------|--------------|
| VO + Sarari | Tabbing   | "Some card"  |
| VO + Sarari | Reading   | "Some card"  |
| VO + Sarari | Link list | "Some card"  |
| NVDA + FF   | Tabbing   | "Some card"  |
| NVDA + FF   | Reading   | "Some card"  |
| JAWS + IE   | Tabbing   | ""           |
| JAWS + IE   | Reading   | "Some card"  |


@straker can you do some more testing for this, complete this table? Tabbing to the link with JAWS does seem to cause problems, so that suggests there is at least some sort of issue here. I want to get a better sense of the problem before we decide whether or not we let axe fail this, warn about it, or let it pass.

**Comment by WilcoFiers** [comment]
@drewlee What element are you using inside links that is causing this behaviour? I think the way to fix it is to not use that element.

**Comment by drewlee** [comment]
@WilcoFiers, that's much easier said than done. The contention seems to be around our usage of a definition list, but unordered/ordered lists are failing as well. From a semantic perspective, our usage is completely justifiable and makes more sense than a hierarchy of divs. I'd also like to point out that in accordance with the HTML5 specification, it is perfectly acceptable to encompass complex semantics within anchors: [https://www.w3.org/TR/html52/textlevel-semantics.html#elementdef-a](). To paraphrase: "the `<a>` element may be wrapped around entire paragraphs, lists, tables, and so forth, even entire sections, so long as there is no interactive content within (e.g., buttons or other links)."

The specific JAWS+IE failure in the matrix above seems like a vendor deficiency. If a user interface is built in accordance to the specification, and a vendor fails to properly implement it, then it's rather heavy handed and overtly opinionated to flag this as a failure for the majority of use cases. I don't think this rule is very effective or useful if it's intent is to force all developers to limit anchor usage to simple text strings.

**Comment by dylanb** [comment]
As sad as this may seem to all the JAWS users out there, it is not required for meeting accessibility supported.

**Comment by padmavemulapati** [comment]
Verified in the latest develop code of axe-core(v4.0.2), 
Test snippet: 
`<a href="...">
  <article><div>
    <div><img src="..." alt=""></div>
    <p class="card-description">Some card</p>
  </div></article>
</a>`
Not seeing any failure for `link-name`
![image](https://user-images.githubusercontent.com/47409579/98975562-4329a400-253c-11eb-9ea5-81e2579c8c27.png)
